esphome/components/esphome/ota/ota_esphome.cpp

```diff
@@ -82,7 +82,14 @@ void ESPHomeOTAComponent::dump_config() {
 #endif
 }
 
-void ESPHomeOTAComponent::loop() { this->handle_(); }
+void ESPHomeOTAComponent::loop() {
+  // Skip handle_() call if no client connected and no incoming connections
+  // This optimization reduces idle loop overhead when OTA is not active
+  // Note: No need to check server_ for null as the component is marked failed in setup() if server_ creation fails
+  if (client_ != nullptr || server_->ready()) {
```

Comment:
It seems we are missing `this->` all over the place in this file. 🙈
```suggestion
  if (this->client_ != nullptr || this->server_->ready()) {
```

Reply:
I think I got them all in ca6ae746c1c53aa74732fe369fe2101d0f3ec196